runtime/v2/shim/util.go

```diff
@@ -85,7 +85,7 @@ func Command(ctx context.Context, config *CommandConfig) (*exec.Cmd, error) {
 func BinaryName(runtime string) string {
 	// runtime name should format like $prefix.name.version
 	parts := strings.Split(runtime, ".")
-	if len(parts) < 2 {
+	if len(parts) < 2 || parts[0] == "" {
```

Comment:
Still not robust to catch a case like `foo/bar`.

Probably, we should just do : 
```go
runtime := clicontext.String("runtime") 
if strings.Contains(runtime, "/") && !filepath.IsAbs(runtime) {
  return fmt.Errorf("runtime must be a runtime name (e.g., \"io.containerd.runc.v2\") or absolute path to the runtime binary (e.g., \"/usr/local/bin/containerd-shim-runc-v2\"), got %q", runtime)
}
```

Reply:
From my testing, the existing check `len(parts) < 2` already covers this case as `strings.Split("foo/bar", ".")` will result in a slice of length 1 : `["foo/bar"]` https://go.dev/play/p/B-JZOTG9cte

Happy to go the direction you suggested if that's preferred, it does seem to more explicitly check for this case 🙂

Reply:
Would you like this to go in [runtime/v2/manager.go](https://github.com/containerd/containerd/blob/2a588a619e76384e7d44e17100b694c728ac5fcf/runtime/v2/manager.go#L237-L246) here ? or would it belong better in [cmd/ctr/commands/run/run_unix.go](https://github.com/containerd/containerd/blob/330961c2d59903c6931ed52f722701fa435e326b/cmd/ctr/commands/run/run_unix.go#L369-L387) ?

Reply:
I prefer to have that under runtime/ rather than cmd/